lms/static/js/Markdown.Editor.js

```diff
@@ -1130,14 +1130,31 @@
             cancelButton.onclick = function() { return close(true); };
 
             if (imageUploadHandler) {
+                okButton.disabled = true;
                 var startUploadHandler = function() {
-                    document.getElementById('file-upload').onchange = function() {
+                    document.getElementById('file-upload').onchange = function(event) {
                         imageUploadHandler(this, urlInput);
                         urlInput.focus();
 
+                        var lengthOfDescriptionField = $.trim(descInput.value).length;
+                        okButton.disabled = !(lengthOfDescriptionField && event.target.value.length);
                         // Ensures that a user can update their file choice.
                         startUploadHandler();
                     };
+
+                    urlInput.oninput = toggleSubmitButton;
+
+                    descInput.oninput = toggleSubmitButton;
+
+                    function toggleSubmitButton() {
+                      var lengthOfDescriptionField = $.trim(descInput.value).length;
+                      var valueOfUrlField = $.trim(urlInput.value);
+                      var lengthOfUrlFieldWithoutProtocol = valueOfUrlField.replace(/^https?:\/\//i, '').length;
+                      var lengthOfFileUploadField = document.getElementById('file-upload').value.length;
+
+                      okButton.disabled = !((lengthOfDescriptionField && lengthOfFileUploadField) ||

```

Comment:
simplify these conditions as these are hard to read. For example:
```
if (uploadFieldHasContent || descriptionFieldHasContent) {
}
```

Reply:
Done

Reply:
Done